pubsub-adapter/src/main/kotlin/io/ably/lib/realtime/WrapperRealtimeClient.kt

```diff
@@ -1,50 +1,87 @@
 package io.ably.lib.realtime
 
-import com.ably.pubsub.*
+import com.ably.pubsub.Channels
+import com.ably.pubsub.RealtimeChannel
+import com.ably.pubsub.RealtimeClient
+import com.ably.pubsub.RealtimePresence
 import com.ably.query.OrderBy
 import io.ably.lib.buildHistoryParams
 import io.ably.lib.http.Http
 import io.ably.lib.types.*
 
-internal class WrapperRealtimeClient(private val javaClient: AblyRealtime, private val httpModule: Http) : RealtimeClient by RealtimeClientAdapter(javaClient) {
-  override val channels: Channels<out RealtimeChannel>
-    get() = WrapperRealtimeChannels(javaClient.channels, httpModule)
+internal class WrapperRealtimeClient(
+    private val javaClient: AblyRealtime,
+    private val httpModule: Http,
+    private val agents: Map<String, String>,
+) : RealtimeClient by RealtimeClientAdapter(javaClient) {
```

Comment:
Does the `request` method need to be overridden here to inject the wrapper SDK agents, or is there some other mechanism that handles this that I haven't understood?

Reply:
I used to patch AblyRealtime client itself, but I agree it's not very obvious.  Now I override all methods instead and do nothing with underlying realtime client